### PR TITLE
Add tests for C++ functional cosine_similarity and pairwise_distance, and clean up functional test code

### PR DESCRIPTION
**Pull request by yf225**
This ensures that `F::cosine_similarity` and `F::pairwise_distance` can be used simply by including `torch/torch.h` and set `namespace F = torch::nn::functional`.

### CODE REVIEW

**Comment by yf225** [comment]
@jon-tow Would you like to also review this PR? My apologies for omitting this when reviewing https://github.com/pytorch/pytorch/pull/26424. Thanks!

**Comment by jon-tow** [comment]
@yf225 Good catch! This looks good to me :+1:. 
One small "nit" question: Do we need the `torch::requires_grad()` option given that we don't check gradient responses here? Or is this more of a convention? 
Thanks!

**Comment by yf225** [comment]
@jon-tow Thanks for the great catch - I just removed the `torch::requires_grad()` option 😄

**Review by pbelevich — APPROVED** [review]

**Review by jon-tow — APPROVED** [review]

File: test/cpp/api/functional.cpp [hunk 1]
```diff
@@ -56,10 +56,26 @@ TEST_F(FunctionalTest, AvgPool2d) {
...
+  auto x = torch::ones({2, 5, 5, 5});
   auto y = F::avg_pool3d(x, AvgPool3dOptions(3).stride(2));
 
   ASSERT_EQ(y.ndimension(), 4);
   ASSERT_TRUE(torch::allclose(y, torch::ones({2, 2, 2, 2})));
   ASSERT_EQ(y.sizes(), torch::IntArrayRef({2, 2, 2, 2}));
 }
+
+TEST_F(FunctionalTest, CosineSimilarity) {
+  auto input1 = torch::tensor({{1, 2, 3}, {4, 5, 6}});
```
Comment (jon-tow) [inline test/cpp/api/functional.cpp]: Looks like we'll need to specify the type as `torch::kFloat` for both `input1` and `input2`. Sorry for not picking this out earlier.

File: test/cpp/api/functional.cpp [hunk 2]
```diff
@@ -56,10 +56,26 @@ TEST_F(FunctionalTest, AvgPool2d) {
...
+  auto input2 = torch::tensor({{1, 8, 3}, {2, 1, 6}});
```
Comment (jon-tow) [inline test/cpp/api/functional.cpp]: As above, we'll need to pass the `torch::kFloat` type.

**Review by jon-tow — CHANGES_REQUESTED** [review]
@yf225 I let a little mistake slip through! I detailed it in the comments below.

File: test/cpp/api/functional.cpp (hunk 1 above)
Comment (yf225) [reply test/cpp/api/functional.cpp]: Thanks a lot for the catch! Just fixed it.

**Review by jon-tow — APPROVED** [review]
